mobile/salto/src-tauri/src/background_tasks.rs

```diff
@@ -0,0 +1,63 @@
+use std::{sync::Arc, time::Duration};
+use tauri::Emitter;
+use tokio::sync::RwLock;
+
+use crate::{PriceConfig, PriceResponce};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Reqwest(#[from] reqwest::Error),
+    #[error(transparent)]
+    Tauri(#[from] tauri::Error),
+}
+
+async fn fetch_and_emit_prices(url: &str, app: &tauri::AppHandle) -> Result<(), Error> {
+    let resp = reqwest::get(url).await?;
+    let new_prices = resp.json::<PriceResponce>().await?;
+    app.emit("new-price", new_prices)?;
```

Comment:
you just need to emit the event for the selected currency. The front don't care about the prices of the currencies that have not been selected by user.